DifferentiationInterface/ext/DifferentiationInterfaceGTPSAExt/twoarg.jl

```diff
@@ -0,0 +1,195 @@
+    for b in eachindex(tx, ty)
+        dx, dy = tx[b], ty[b]
+        foreach((t, xi, dxi) -> (t[0] = xi; t[1] = dxi), prep.xt, x, dx)
+        fc!(prep.yt, prep.xt)
+        map!(t -> t[1], dy, prep.yt)
+    end
+    map!(t -> t[0], y, prep.yt)
+    return ty
+end
+
+function DI.value_and_pushforward(
+    f!,
+    y,
+    prep::GTPSATwoArgPushforwardPrep,
+    backend::AutoGTPSA,
+    x,
+    tx::NTuple,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    ty = DI.pushforward(f!, y, prep, backend, x, tx, contexts...)
+    return y, ty
+end
+
+function DI.value_and_pushforward!(
+    f!,
+    y,
+    ty::NTuple,
+    prep::GTPSATwoArgPushforwardPrep,
+    backend::AutoGTPSA,
+    x,
+    tx::NTuple,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    DI.pushforward!(f!, y, ty, prep, backend, x, tx, contexts...)
+    return y, ty
+end
+
+## Jacobian
+# Input: Contains a vector of pre-allocated TPSs
+# Output: Contains a vector of pre-allocated TPSs
+struct GTPSATwoArgJacobianPrep{X,Y} <: DI.JacobianPrep
+    xt::X
+    yt::Y
+end
+
+function DI.prepare_jacobian(
+    f, y, backend::AutoGTPSA{D}, x, contexts::Vararg{DI.Context,C}
+) where {D,C}
+    if D != Nothing
+        d = backend.descriptor
+    else
+        d = Descriptor(length(x), 1) # n variables to first order
+    end
+
+    # We set the slopes of each variable to 1 here, this will always be the case for Jacobian
+    xt = similar(x, TPS{promote_type(eltype(x), Float64)})
+    j = 1
+    for i in eachindex(xt)
+        xt[i] = TPS{promote_type(eltype(x), Float64)}(; use=d)
+        xt[i][j] = 1
+        j += 1
+    end
+
+    yt = similar(y, TPS{promote_type(eltype(y), Float64)})
+
+    for i in eachindex(yt)
+        yt[i] = TPS{promote_type(eltype(y), Float64)}(; use=d)
+    end
+
+    return GTPSATwoArgJacobianPrep(xt, yt)
+end
+
+function DI.jacobian(
+    f!, y, prep::GTPSATwoArgJacobianPrep, ::AutoGTPSA, x, contexts::Vararg{DI.Context,C}
+) where {C}
+    foreach((t, xi) -> t[0] = xi, prep.xt, x) # Set the scalar part
+    fc! = DI.with_contexts(f!, contexts...)
+    fc!(prep.yt, prep.xt)
+    jac = similar(x, GTPSA.numtype(eltype(prep.yt)), (length(prep.yt), length(x)))
+    GTPSA.jacobian!(jac, prep.yt; include_params=true, unsafe_inbounds=true)
+    map!(t -> t[0], y, prep.yt)
+    return jac
+end
+
+function DI.jacobian!(
+    f!,
+    y,
+    jac,
+    prep::GTPSATwoArgJacobianPrep,
+    ::AutoGTPSA,
+    x,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    foreach((t, xi) -> t[0] = xi, prep.xt, x) # Set the scalar part
+    fc! = DI.with_contexts(f!, contexts...)
+    fc!(prep.yt, prep.xt)
+    GTPSA.jacobian!(jac, prep.yt; include_params=true, unsafe_inbounds=true)
+    map!(t -> t[0], y, prep.yt)
+    return jac
+end
+
+function DI.value_and_jacobian(
+    f!,
+    y,
+    prep::GTPSATwoArgJacobianPrep,
+    backend::AutoGTPSA,
+    x,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    jac = DI.jacobian(f!, y, prep, backend, x, contexts...)
```

Comment:
How do we ensure that `y` contains the right thing at the end of `GTPSA.jacobian!`?